cwms-data-api/src/main/java/cwms/cda/data/dao/LocationLevelsDaoImpl.java

```diff
@@ -623,7 +622,7 @@ private static TimeSeries buildTimeSeries(ILocationLevelRef levelRef, Interval i
             if (qualityCode != null) {
                 quality = qualityCode.intValue();
             }
-            timeSeries.addValue(dateTime, value, quality);
+            timeSeries.addValue(dateTime, value, quality, null);
```

Comment:
Are we sure this isn't a breaking change?

Reply:
I'll double check with additional test cases, but this change should not break any existing functionality. A null data entry date parameter is treated as if a standard Time-Value-Quality data entry was provided. The implementation of `addValue` will use a TimeSeries data record with only three input fields under normal circumstances: 

` if (dataEntryDate != null) {
                values.add(new TimeSeriesRecordWithDate(dateTime, value, qualityCode, dataEntryDate));
            } else {
                values.add(new Record(dateTime, value, qualityCode));
            }`

The existing use cases of TimeSeries should be unaffected, as they will be handled exactly as they were before these changes.

Reply:
I recommend subclassing TimeSeries instead

Reply:
I think it would break CWMS.js, The javascript openapi generator already has trouble with our TimeSeries class given some specific assumptions the generator chose to make.

Reply:
> I think it would break CWMS.js, The javascript openapi generator already has trouble with our TimeSeries class given some specific assumptions the generator chose to make.

Just chiming back in here to add some context... We're already modifying the OpenAPI spec in the cwmsjs build process to adjust a few things, so we can just handle this as well if need be.  I don't think it will cause an issue (not a new one, at least), though, because at this point cwmsjs just anticipates an array of numbers in the response and leaves it up to the user to determine what is what based on either the request or the response metadata.